reproduce-figures/prepare-scpca-portal-data.R

```diff
@@ -0,0 +1,309 @@
+fs::dir_create(dirname(output_merged_sce_path))
+fs::file_copy(
+  opts$merged_sce_path, 
+  output_merged_sce_path
+)
+
+
+# Map over project zip files to organize files for reproducibility ------------------
+input_zips |>
+  purrr::iwalk(
+    \(project_zip, project_id){
+      
+      # temporary variables used for testing
+      #project_id <- "SCPCP000001"
+      #project_zip <- "/Users/sjspielman/Downloads/projects/SCPCP000001_SINGLE-CELL_SINGLE-CELL-EXPERIMENT_2025-04-03.zip"
+      
+      project_consensus_dir <- file.path(output_dir_consensus_files, project_id)
+      
+      # Unzip the directory, specifying the target in consensus
+      unzip(project_zip, exdir = project_consensus_dir)
+
+      # First, remove all the qc htmls
+      system(glue::glue("rm {project_consensus_dir}/*/*html"))
+      
+      # Define all sample directories present in this project, _excluding_ multiplexed samples
+      # This is because multiplexed samples aren't used in any figures created from ScPCA data files
+      sample_dirs <- list.dirs(
+        project_consensus_dir, 
+        full.names = TRUE
+      ) |>
+        purrr::set_names(\(x) {basename(x)}) |>
+        # get rid of the project folder & multiplexed samples
+        purrr::discard_at(\(x) {stringr::str_detect(x, "SCPCP\\d{6}")}) |>
+        purrr::discard_at(\(x) {stringr::str_detect(x, "_")}) 
+
+      ####### Step 1: Copy RDS files to target directories #######
+      sample_dirs |>
+        purrr::iwalk(
+          \(sample_dir, sample_id) {
+            
+            # Copy processed rds files to celltype_results_files
+            # The next step will parse these files into TSVs
+            if (sample_id %in% celltype_results_samples) {
+              system(glue::glue("cp {sample_dir}/*processed.rds {output_dir_celltype_results}"))
+            } 
+            
+            # Copy rds files to standalone sample dir
+            if (sample_id %in% standalone_sample_dirs) {
+              system(glue::glue("cp -r {sample_dir} {output_dir_s3_files}"))
+            } 
+          
+            # We can now remove the (un)filtered files
+            system(glue::glue("rm {sample_dir}/*filtered.rds"))
+      })
+
+      # If this project is used in the bulk analysis,copy project over as well
+      if (project_id %in% bulk_projects) {
+        fs::dir_copy(
+          project_consensus_dir, 
+          output_dir_analysis_data
+        )
+      }
+      
+      ####### Step 2: Prepare consensus cell type files #######
+      sample_dirs |>
+        purrr::walk(
+          \(sample_dir) {
+              
+            # Define input and output files
+            processed_rds_file <- list.files(
+              path = sample_dir,
+              pattern = "_processed\\.rds$", 
+              full.names = TRUE
+            )
+            stopifnot("Expected a single processed rds file." = length(processed_rds_file) == 1)
```

Comment:
What about when there are multiple libraries per sample? Then we have two processed files.

Reply:
Looking through our library metadata right now, and I'm not sure what circumstance this would occur in _besides_ multiplexed, which again as I am understanding things, we don't need data for and has already been removed (am I correct though?). 

When I look at duplicate library ids, it's always 10x & cite, or 10x & cellhash, and 10x & cite is still only 1 processed file. So, I'm not sure what scenario we'd expect more than 1 here. What am I missing?

Reply:
> When I look at duplicate library ids, it's always 10x & cite, or 10x & cellhash, and 10x & cite is still only 1 processed file. So, I'm not sure what scenario we'd expect more than 1 here. What am I missing?

I'm not talking about duplicate library IDs, I'm talking about the scenario where we have multiple unique libraries per sample. There's not always a 1:1 ratio for sample to library. Here you are getting all `_processed.rds` files in a sample directory and if we have multiple libraries with the same sample ID then you will get multiple `_processed.rds` files that each will need to be read in to grab consensus cell types.